apps/saved_searches/saved_searches.py

```diff
@@ -199,6 +201,19 @@ def on_fetched(self, docs):
     def on_deleted(self, doc):
         push_notification(UPDATE_NOTIFICATION)
 
+    def on_delete(self, doc):
+        self._validate_user(doc['user'])
+
+    def _validate_user(self, doc_user_id, create=False):
+        session_user = get_user(required=True)
+        if create and str(session_user['_id']) != doc_user_id:
+            raise SuperdeskApiError.forbiddenError(
+                'Unauthorized to create another user\'s search.')
+
+        if str(session_user['_id']) != doc_user_id and not current_user_has_privilege('global_saved_searches') and \
```

Comment:
`current_user_has_privilege` function checks if the user is admin so we don't need to check it here again...